modules/features2d/src/orb.cpp

```diff
@@ -650,6 +650,19 @@ static inline float getScale(int level, int firstLevel, double scaleFactor)
     return (float)std::pow(scaleFactor, (double)(level - firstLevel));
 }
 
+static inline int getLevel(KeyPoint &kp)
+{
+    int level = kp.octave;
+    CV_Assert(0 <= level);
+    //SIFT encodes layer and octave together
```

Comment:
I see here this design problem:
- ORB should not have any knowledge about special cases of other descriptors/detectors, like SIFT.

We can try placing this logic into KeyPoint directly (some unified behavior with documentation).

Reply:
@alalek
- Can we introduce the attribute `KeyPoint.layer`?
- Can we allow `KeyPoint.octave` to have a negative value?